src/ResourceManagement/Sql/SqlDatabaseAutomaticTuningImpl.cs

```diff
@@ -0,0 +1,156 @@
+        private string sqlDatabaseName;
+        private Dictionary<string,Models.AutomaticTuningOptions> automaticTuningOptionsMap;
+
+        ///GENMHASH:0B123513DA66B23285D7B0CA6EE63B8D:7ADD22B33B14B9D9F2AB7F5856755122
+        internal SqlDatabaseAutomaticTuningImpl(SqlDatabaseImpl database, DatabaseAutomaticTuningInner innerObject)
+            : base(innerObject)
+        {
+            this.sqlServerManager = database.sqlServerManager;
+            this.resourceGroupName = database.resourceGroupName;
+            this.sqlServerName = database.sqlServerName;
+            this.sqlDatabaseName = database.Name();
+            this.key = Guid.NewGuid().ToString();
+            this.automaticTuningOptionsMap = (innerObject != null && innerObject.Options != null) ? new Dictionary<string, AutomaticTuningOptions>(innerObject.Options) : new Dictionary<string, AutomaticTuningOptions>();
+        }
+
+        ///GENMHASH:7DFD4E2C4BFE1C1A223F1E9652CEA12E:095451E62832B6CE321960A429995E7F
+        internal SqlDatabaseAutomaticTuningImpl(string resourceGroupName, string sqlServerName, string sqlDatabaseName, DatabaseAutomaticTuningInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject)
+        {
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            this.sqlDatabaseName = sqlDatabaseName;
+            this.key = Guid.NewGuid().ToString();
+            this.automaticTuningOptionsMap = (innerObject != null && innerObject.Options != null) ? new Dictionary<string, AutomaticTuningOptions>(innerObject.Options) : new Dictionary<string, AutomaticTuningOptions>();
+        }
+
+        ///GENMHASH:17950B8F2EC46B114539E23F9F4D3EA2:EBF69F3A6BD7C5BE989345C27B59C5FE
+        public AutomaticTuningMode DesiredState()
+        {
+            return this.Inner.DesiredState.GetValueOrDefault();
+        }
+
+        ///GENMHASH:93F5525F475C77754B229151C3005F4B:FE981C7CC77129BC45DD1FFCFCD2402F
+        public async Task<Microsoft.Azure.Management.Sql.Fluent.ISqlDatabaseAutomaticTuning> ApplyAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            this.Inner.Options = this.automaticTuningOptionsMap.Count == 0 ? null : this.automaticTuningOptionsMap;
+            var databaseAutomaticTuningInner = await this.sqlServerManager.Inner.DatabaseAutomaticTuning
+                .UpdateAsync(this.resourceGroupName, this.sqlServerName, this.sqlDatabaseName, this.Inner, cancellationToken);
+            this.SetInner(databaseAutomaticTuningInner);
+            this.automaticTuningOptionsMap = (databaseAutomaticTuningInner != null && databaseAutomaticTuningInner.Options != null) ? new Dictionary<string, AutomaticTuningOptions>(databaseAutomaticTuningInner.Options) : new Dictionary<string, AutomaticTuningOptions>();
+            return databaseAutomaticTuningInner != null ? this : null;
```

Comment:
I'm not sure I understood what are you trying to do here. Why would you return null in any case within ApplyAsync? Throwing exception on error should be preferred way, right?

Reply:
Fixed